mne/io/meas_info.py

```diff
@@ -157,7 +248,83 @@ def copy(self):
         info : instance of Info
             The copied info.
         """
-        return Info(super(Info, self).copy())
+        return Info(self._store)
+
+    def to_dict(self):
+        """Obtain a version of this info object in the form of a plain Python
+        dict
+
+        While the Info object behaves like a dict, it is a subclass of
+        MutableMapping. Use this function to cast it to an actual dict. Be
+        aware that the read-only fields will become writable and will no longer
+        auto-update.
+
+        Notes
+        -----
+        - To cast the plain dict back to an instance of Info, use:
+          `mne.io.Info(info_dict)`
+        - Always use a propert instance of Info when calling MNE-Python
+          functions.
+        """
+        info_dict = dict(self)
+        # Convert non-standard fields
+        info_dict['ch_names'] = list(info_dict['ch_names'])
```

Comment:
yes. it iterates over the items when constructing the new dict. and nchan and ch_names are listed in the items.